keras_cv/layers/preprocessing3d/global_random_scaling.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing3d import base_augmentation_layer_3d
+
+POINT_CLOUDS = base_augmentation_layer_3d.POINT_CLOUDS
+BOUNDING_BOXES = base_augmentation_layer_3d.BOUNDING_BOXES
+
+
+class GlobalRandomScaling(base_augmentation_layer_3d.BaseAugmentationLayer3D):
+    """A preprocessing layer which randomly scales point clouds and bounding boxes along
+    X, Y, and Z axes during training.
+
+    This layer will randomly scale the whole scene along the  X, Y, and Z axes based on a randomly sampled
+    scaling factor between [min_scaling_factor, max_scaling_factor] following a uniform distribution.
+    During inference time, the output will be identical to input. Call the layer with `training=True` to scale the input.
+
+    Input shape:
+      point_clouds: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of points, num of point features].
+        The first 5 features are [x, y, z, class, range].
+      bounding_boxes:  3D (multi frames) float32 Tensor with shape
+        [num of frames, num of boxes, num of box features].
+        The first 7 features are [x, y, z, dx, dy, dz, phi].
+
+    Output shape:
+      A dictionary of Tensors with the same shape as input Tensors.
+
+    Arguments:
+      scaling_factor_x: A tuple of float scalar sets the minimum and maximum scaling factors for the X axis.
+      scaling_factor_y: A tuple of float scalar sets the minimum and maximum scaling factors for the Y axis.
+      scaling_factor_z: A tuple of float scalar sets the minimum and maximum scaling factors for the Z axis.
+    """
+
+    def __init__(
+        self,
+        scaling_factor_x,
```

Comment:
(here and in the other KPLs where you've added defaults) -- let's make this default to None in the constructor like

`scaling_factor_x=None`

Reply:
Done.